pyfar/dsp/dsp.py

```diff
@@ -1395,6 +1395,119 @@ def time_shift(signal, shift, unit='samples'):
+        response.
+        The fft_norm of this resulting signal is set to 'none'.
+
+    References
+    -----------
+    .. [#] S. Mueller and P. Masserani "Transfer function measurement with
+           sweeps. Directors cut." J. Audio Eng. Soc. 49(6):443-471,
+           (2001, June).
+    """
+
+    # Check if system_output and system_input are both type Signal
+    if not isinstance(system_output, pyfar.Signal):
+        raise TypeError('system_output has to be of type pyfar.Signal')
+    if not isinstance(system_input, pyfar.Signal):
+        raise TypeError('system_input has to be of type pyfar.Signal')
+
+    # Check if both signals have the same sampling rate
+    if not system_output.sampling_rate == system_input.sampling_rate:
+        raise ValueError("The two signals have different sampling rates!")
+
+    # Set fft_length to the max n_samples of both signals,
+    # if it is not explicitly set to a value
+    if fft_length is None:
+        fft_length = np.max([system_output.n_samples, system_input.n_samples])
+    # Check if both signals length are shorter or the same as fft_length
+    if fft_length < system_output.n_samples:
+        raise ValueError("The fft_length can not be shorter than" +
+                         "system_output.n_samples.")
+    if fft_length < system_input.n_samples:
+        raise ValueError("The fft_length can not be shorter than" +
+                         "system_input.n_samples.")
+
+    # Check if both signals have the same length as ftt_length,
+    # if not: bring them to the same length by padding with zeros
+    if system_output.n_samples < fft_length:
+        # Add Zeros to system_input
+        system_output = pyfar.dsp.pad_zeros(system_output,
+                                            (fft_length -
+                                             system_output.n_samples))
+
+    if system_input.n_samples < fft_length:
+        # Add Zeros to system_output
+        system_input = pyfar.dsp.pad_zeros(system_input,
+                                           (fft_length -
+                                            system_input.n_samples))
+
+    # multiply system_output signal with regularized inversed system_input
+    # signal to get the system response
+    system_response = (system_output *
+                       regularized_spectrum_inversion(system_input, **kwargs))
+
+    # Check if the signals have any comments,
+    # if yes: concatenate the comments for the system_response
+    system_response.comment = "Calculated with pyfar.dsp.deconvolve."
+    if system_output.comment != 'none':
+        system_response.comment += f" system input: {system_output.comment}."
+    if system_input.comment != 'none':
+        system_response.comment += f" system output: {system_input.comment}."
+
+    # return the impulse resonse
+    system_response.fft_norm = pyfar.classes.audio._match_fft_norm(
+        system_output.fft_norm, system_input.fft_norm)
```

Comment:
Deconvolution means division in frequency domain, therefore:
```suggestion
    system_response.fft_norm = pyfar.classes.audio._match_fft_norm(
        system_output.fft_norm, system_input.fft_norm, division=True)
```
Adding a small test to check for the correct call might make sense here :)

Reply:
You are totally right, I thought that the `division=True` will be handled in the `regularized_spectrum_inversion`, but as the resulting fft_norm is determined directly from the parameters of this function, your suggestion is the right way of handling the fft_norm in this case.
Thanks for pointing this out. :)